sunbeam-python/sunbeam/provider/maas.py

```diff
@@ -397,4 +415,131 @@ def list_networks_cmd(format: str):
             table.add_row(network, space or "[italic]<unmapped>[italic]")
         console.print(table)
     elif format == FORMAT_YAML:
-        console.print(mapping, end="")
+        console.print(yaml.dump(mapping), end="")
+
+
+def _run_maas_checks(checks: list[DiagnosticsCheck], console: Console) -> list[dict]:
+    """Run checks sequentially.
+
+    Runs each checks, logs whether the check passed or failed.
+    Prints to console every result.
+    """
+    check_results = []
+    OK = "[green]OK[/green]"
+    FAIL = "[red]FAIL[/red]"
+    for check in checks:
+        LOG.debug(f"Starting check {check.name!r}")
+        message = f"{check.description}..."
+        with console.status(message):
+            results = check.run()
+            if not results:
+                raise ValueError(f"{check.name!r} returned no results.")
+
+            if isinstance(results, DiagnosticsResult):
+                results = [results]
+
+            for result in results:
+                LOG.debug(f"{result.name=!r}, {result.passed=!r}, {result.message=!r}")
+                console.print(
+                    message,
+                    result.message,
+                    "-",
+                    OK if result.passed else FAIL,
+                )
+                check_results.append(result.to_dict())
+    return check_results
+
+
+def _run_maas_meta_checks(
+    checks: list[DiagnosticsCheck], console: Console
+) -> list[dict]:
+    """Run checks sequentially.
+
+    Runs each checks, logs whether the check passed or failed.
+    Only prints to console last check result.
+    """
+    check_results = []
+    OK = "[green]OK[/green]"
+    FAIL = "[red]FAIL[/red]"
```

Comment:
can we push these two constants out of functions and make then globally re-usable?

Reply:
Fixed in last commit